Source/JavaScriptCore/inspector/agents/InspectorConsoleAgent.cpp

```diff
@@ -252,15 +252,14 @@ void InspectorConsoleAgent::addConsoleMessage(std::unique_ptr<ConsoleMessage> co
         if (m_enabled)
             previousMessage->updateRepeatCountInConsole(*m_frontendDispatcher);
     } else {
-        ConsoleMessage* newMessage = consoleMessage.get();
-        m_consoleMessages.append(WTFMove(consoleMessage));
         if (m_enabled) {
             auto generatePreview = !m_isAddingMessageToFrontend;
             SetForScope isAddingMessageToFrontend(m_isAddingMessageToFrontend, true);
 
-            newMessage->addToFrontend(*m_frontendDispatcher, m_injectedScriptManager, generatePreview);
+            consoleMessage->addToFrontend(*m_frontendDispatcher, m_injectedScriptManager, generatePreview);
         }
 
+        m_consoleMessages.append(WTFMove(consoleMessage));
```

Comment:
actually i dont think we need to save the `ConsoleMessage` if the `InspectorConsoleAgent` is `m_enabled` since we're immediately sending it to the frontend.  in that case we only really need the `m_consoleMessages.last().get()` to check if the new `consoleMessage` matches the same type, value, etc., so we should either have a separate member variable for `m_lastConsoleMessage` or just replace `m_consoleMessages.last() = WTFMove(consoleMessage)` (maybe with an `ASSERT(m_consoleMessages.size() == 1)`).
```
if (!m_consoleMessages.isEmpty()) {
    auto& previousMessage = m_consoleMessages.last();
    if (!isGroupMessage(previousMessage.type()) && previousMessage.isEqual(consoleMessage.get())) {
        previousMessage->incrementCount();
        if (m_enabled)
            previousMessage->updateRepeatCountInConsole(*m_frontendDispatcher);
    }
    return;
}

if (!m_enabled) {
    m_consoleMessages.append(WTFMove(consoleMessage));
    if (m_consoleMessages.size() >= maximumConsoleMessages) {
        m_expiredConsoleMessageCount += expireConsoleMessagesStep;
        m_consoleMessages.remove(0, expireConsoleMessagesStep);
    }
    return;
}

auto generatePreview = !m_isAddingMessageToFrontend;
SetForScope isAddingMessageToFrontend(m_isAddingMessageToFrontend, true);

newMessage->addToFrontend(*m_frontendDispatcher, m_injectedScriptManager, generatePreview);

ASSERT(m_consoleMessages.size() <= 1);
if (m_consoleMessages.size() < 1)
    m_consoleMessages.append(WTFMove(consoleMessage));
else
    m_consoleMessages.last() = WTFMove(consoleMessage);
```

Reply:
This is assumedly a cherry pick, so not sure why folks were pinged for review 😅. But I think we do want to store the messages still so that if a front end is closed and later reopened beyond the 1 or 2 minute cache/keep alive time on the frond end we will still have a record of the last `maximumConsoleMessages` messages.

Reply:
oh interesting i didn't realize it's a cherry pick

hmm i suppose yeah that's a reasonable reason to keep them (or at least the last ~100 messages)

might be worth a comment explaining that so future readers aren't left wondering "why do we still need to keep these if we've already sent them to the frontend"